journal.py

```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import os
+import datetime
+from pyramid.config import Configurator
+from pyramid.view import view_config
+from waitress import serve
+
+import sqlalchemy as sa
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import scoped_session, sessionmaker
+from zope.sqlalchemy import ZopeTransactionExtension
+
+from pyramid.httpexceptions import HTTPFound
+from sqlalchemy.exc import DBAPIError
+
+from pyramid.authentication import AuthTktAuthenticationPolicy
+from pyramid.authorization import ACLAuthorizationPolicy
+from cryptacular.bcrypt import BCRYPTPasswordManager
+from pyramid.security import remember, forget
+
+Base = declarative_base()
+DBSession = scoped_session(sessionmaker(extension=ZopeTransactionExtension()))
+DATABASE_URL = os.environ.get(
+    'DATABASE_URL',
+    'postgresql://jimgrant@localhost:5432/learning-journal'
+)
+HERE = os.path.dirname(os.path.abspath(__file__))
+
+
+class Entry(Base):
+    __tablename__ = 'entries'
+
+    id = sa.Column(sa.Integer, primary_key=True, autoincrement=True)
+    title = sa.Column(sa.Unicode(127), nullable=False)
+    text = sa.Column(sa.UnicodeText, nullable=False)
+    created = sa.Column(
+        sa.DateTime, nullable=False, default=datetime.datetime.utcnow
+    )
+
+    def __repr__(self):
+        return self.title
+
+    @classmethod
+    def all(cls, session=None):
+        if session is None:
+            session = DBSession
+        return session.query(cls).order_by(cls.created.desc()).all()
+
+    @classmethod
+    def write(cls, title=None, text=None, session=None):
+        if session is None:
+            session = DBSession
+        instance = cls(title=title, text=text)
+        session.add(instance)
+        return instance
+
+    @classmethod
+    def one(cls, eid=None, session=None):
+        if session is None:
+            session = DBSession
+        return session.query(cls).filter(cls.id == eid).one()
+
+    @classmethod
+    def modify(cls, eid=None, title=None, text=None, session=None):
+        if session is None:
+            session = DBSession
+        instance = cls.one(eid)
+        instance.title = title
+        instance.text = text
+        session.add(instance)
```

Comment:
This is not needed.  Because you queried the session in order to get the instance returned by `cls.one(eid)`, the entry is already known to the session.  Simply updating the `title` and `text` attributes will be enough that the session will know the object is *dirty*.  Then when the session commits, an "update" query will be emitted.